src/acoupi/system/lifecycle.py

```diff
@@ -37,6 +37,7 @@ def stop_program(settings: Settings) -> None:
     services.stop_services(settings)
     services.disable_services(settings)
     deployment = end_deployment(settings)
+    programs.end_program(settings)
     program = programs.load_program(settings)
     program.on_end(deployment)
```

Comment:
Another thing we could try is to modify the stop services function so that there is some waiting time to finish processing the pending tasks. This could involve stoping the beat service first and waiting for some graceful stopping period until finally stoping the acoupi service with the celery workers.

Reply:
Hmm. Yes, that could work too. Also, I am little unsure what would happen if a task has been registered but has a frequency of xx minutes or even hours. 

Also, if for some reasons there are some pending files in the /tmp-audio, it would be good to manage them at the end of a deployment. In the case a user end a deployment and shutdown the rpi to move to another location, the pending files in /tmp-audio won't be lost.

Reply:
Agree! I think doing both makes sense.